### ISSUE TITLE
Support Shell Integration For Nushell

### ISSUE THREAD

**Issue by Ktoks** [alexpasmantier/television#181]
**Problem**
I would like to propose Nushell shell integration be supported.

**Feature**
Ctrl-r, etc. . . features be supported - similarly to fzf.

**Examples**
Ctrl-r - allows the user to search past command uses in $HOME/.config/nushell

**Additional context**
I don't know that there is additional context needed.

**Comment by Ktoks** [comment]
After using this for a while with Bash and ZSH - I have noticed behavior that leads me to believe that when CTRL-R is pressed - it's just running TV directly on a channel - instead of implementing CTRL-R's behavior - similar to the way FZF or Bash/ZSH/Nushell's default CTRL-R works.

I think this functionality could be more broadly used if it were set up in a way where the user can begin typing a command, hit CTRL-R - an immediately look at the user's current shell history.

As TV stands now - if the user types a partial command and hits CTRL-R - TV fails when you don't have a channel set up for the partial command in the terminal. To get the CTRL-R functionality, the prompt must be empty - or the user has to set up every possible configuration to get the same functionality FZF provides.

Do you want me to put in a separate issue for this?

**Comment by alexpasmantier** [comment]
The behaviour you're describing is I believe the one bound to C-t, but C-r should behave as a regular shell history command. If that's not the case, it's not intended and I'd need to investigate why that happens.

**Comment by alexpasmantier** [comment]
(sorry for the terrible gif image quality)

![cmd_history-ezgif com-optimize](https://github.com/user-attachments/assets/55914b46-0c26-4d1a-9cfc-3601e56be88c)

**Comment by Ktoks** [comment]
You're right - it's when I use C-t without having a channel for the partial command. 
My mistake. I am unsure how I got confused yesterday morning.

**Comment by alexpasmantier** [comment]
No problem, I feel relieved 😊

**Comment by alexpasmantier** [comment]
@Ktoks @GhisF @SrEstegosaurio @askeladd123

Just opened a PR that adds support for NuShell, if you're interested in testing / tweaking / contributing to it, please feel free to do so 😊

**Comment by gzusseat** [comment]
> The behaviour you're describing is I believe the one bound to C-t, but C-r should behave as a regular shell history command. If that's not the case, it's not intended and I'd need to investigate why that happens.

Any idea why calling the Nushell history command from a channel doesn't work here but using the Nushell open command and selecting the command_line does? (sqlite)

**Comment by Ktoks** [comment]
> [@Ktoks](https://github.com/Ktoks) [@GhisF](https://github.com/GhisF) [@SrEstegosaurio](https://github.com/SrEstegosaurio) [@askeladd123](https://github.com/askeladd123)
> 
> Just opened a PR that adds support for NuShell, if you're interested in testing / tweaking / contributing to it, please feel free to do so 😊

I finally got around to testing this.

I think it's close - I have a comment I'll add to the PR.

**Comment by Ktoks** [comment]
I've edited the documentation page - https://github.com/alexpasmantier/television/wiki/Shell-Autocompletion#nushell

Please merge the PR when you have time!

**Comment by alexpasmantier** [comment]
Thanks for the docs and the testing, merged the PR 🎉 

Closing this 😊
